src/connection.rs

```diff
@@ -265,6 +266,7 @@ pub struct ConnectionSender<Exe: Executor> {
 
 impl<Exe: Executor> ConnectionSender<Exe> {
     pub(crate) fn new(
+        connection_id: i64,
```

Comment:
Just a minor recommendation: connection id could be u64 for clarirty?

Reply:
Let's use UUID